R/ei_rxc.R

```diff
@@ -99,114 +113,219 @@ ei_rxc <- function(
   if (verbose) {
     message("Collecting samples...")
   }
-  # Bayes model estimation
-  suppressWarnings(
-    md_out <- ei.MD.bayes(
-      formula = formula,
-      sample = samples,
-      data = data,
-      total = totals_col,
-      thin = thin,
-      burnin = burnin,
-      ret.mcmc = TRUE,
-      tune.list = tune_nocov,
-      ...
+
+  if (diagnostic) {
+    if (verbose) message("Running diagnostic")
+    # Preparation for parallel processing if user specifies parallelization
+    if (par_compute) {
+      # Detect the number of cores you have
+      parallel::detectCores()
+
+      if (parallel::detectCores() < 4) {
+        stop("It is not recommended to run parallel ei with less than 4 cores")
+      } else if (parallel::detectCores() == 4) {
+        warning("4 cores is the minimum recommended to run parallel ei")
+      }
+      if (verbose) message("Running in paralllel")
+
+      # Standard to use 1 less core for clusters
+      clust <- parallel::makeCluster(parallel::detectCores() - 1)
+
+      # Register parallel processing cluster
+      doSNOW::registerDoSNOW(clust)
+
+      # Check to make sure that cores are set up correctly
+      foreach::getDoParWorkers()
+    }
+    # Set infix option for parallel or not parallel
+    `%myinfix%` <- ifelse(par_compute, `%dopar%`, `%do%`)
+
+    # Init progressbar
+    pb <- utils::txtProgressBar(
+      min = 0,
+      max = n_chains,
+      style = 3
     )
-  )
+    progress <- function(n) setTxtProgressBar(pb, n)
+    opts <- list(progress = progress)
 
-  # Extract district-level MCMC chains
-  # These initially present raw population count estimates
-  chains_raw <- md_out$draws$Cell.counts
+    md_mcmc <- foreach::foreach(
+      chain = 1:n_chains,
```

Comment:
I recently learned from pratik that `seq_len(n_chains)` is preferred style over `1:n_chains`. Who knew!